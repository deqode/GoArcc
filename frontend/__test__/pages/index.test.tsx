import renderer from 'react-test-renderer'
import Landing, { handler } from '../../pages'

jest.mock('../../api/rest/fetchUrls', () => {
  return {
    getLoginURL: jest.fn().mockImplementation(() =>
      Promise.resolve({
        url: 'mockUrl',
        error: false,
      })
    )
  }
})



describe('Landing page tests', () => {

  it('should match the snapshot', () => {
    const tree = renderer.create(<Landing url={'moc url'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should get login url if user session not available', async () => {
    const req = {
      session: { get: jest.fn(() => undefined), save: jest.fn() },
    }
    const response = await handler({ req })
    expect(response).toStrictEqual({ props: { url: 'mockUrl' } })
  })
  it('should redirect to dashboard if user session available', async () => {
    const req = {
      session: { get: jest.fn().mockImplementation(() => true), save: jest.fn() },
    }
    const response = await handler({ req })
    expect(response).toStrictEqual({ redirect: { permanent: false, destination: '/dashboard' } })
  })

  it('should redirect to error page if network error', async() => {
    
  })

  it('should redirect to github sign up page on Sing Up button click', async() => {
    
  })
  
})
