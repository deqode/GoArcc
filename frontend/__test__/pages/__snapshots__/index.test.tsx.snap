// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page tests should match the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <h1
          className="MuiTypography-root MuiTypography-h3 MuiTypography-colorPrimary"
        >
          Welcome to Alfred
        </h1>
        <br />
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
        >
          Provide your repo, Select your cloud, and let Alfred do the heavy lifting 
        </h5>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <p
          className="MuiTypography-root makeStyles-text-3 MuiTypography-body1 MuiTypography-displayInline"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            href="moc url"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-label"
            >
              SignUP
            </span>
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
